paddle/framework/lod_tensor.cc

```diff
@@ -103,5 +112,139 @@ void LoDTensor::ShrinkInLevel(size_t level, size_t elem_begin,
   lod_ = new_lod;
 }
 
+std::string LoDTensor::SerializeToString() const {
+  LoDTensorProto desc;
+
+  // set data_type
+  if (this->type() == typeid(int8_t)) desc.set_data_type(DataType::BOOL);
+  if (this->type() == typeid(int16_t)) desc.set_data_type(DataType::INT16);
+  if (this->type() == typeid(int32_t)) desc.set_data_type(DataType::INT32);
+  if (this->type() == typeid(int64_t)) desc.set_data_type(DataType::INT64);
+  // FIXME(dzh): there is no fp16 in standard c++
```

Comment:
```c++
map<string, DataType> typeid2datatype = {
  {typeid(int8_t), DataType::BOOL},
  {typeid(int16_t), DataType::INT16},
  ....
};

desc.set_data_type(typeid2datatype[this->type()]);
```
put this map outside somewhere, may be a member of a struct. 

This can be reused, and make code cleaner.